File: data/TiendaMusical/Dependencies/Cliente.js
/**
 * Clase que representa un cliente de la tienda.
 */
class Cliente {
    /**
     * @param {string} id 
     * @param {string} nombre 
     */
    constructor(id, nombre) {
        this.id = id;
        this.nombre = nombre;
        /** 
         * Historial de compras del cliente.
         * @type {Object.<string, number>} 
         * Clave: ID del producto, Valor: cantidad comprada
         */
        this.historialCompras = {};
    }

    /**
     * Añade una compra al historial del cliente.
     * Valida que la cantidad sea positiva antes de añadirla.
     * @param {string} productoId ID del producto comprado
     * @param {number} cantidad Cantidad comprada
     */
    añadirCompra(productoId, cantidad) {
        if (cantidad > 0) {
            if (this.historialCompras[productoId]) {
                this.historialCompras[productoId] += cantidad;
            } else {
                this.historialCompras[productoId] = cantidad;
            }
        }
    }

    /**
     * Calcula el total gastado por el cliente.
     * @param {Object.<string, Producto>} catalogo Catalogo de productos. 
     *        Clave: ID del producto, Valor: objeto Producto con el precio.
     * @return {number} Total gastado
     */
    calcularTotalGastado(catalogo) {
        let total = 0;
        for (const [productoId, cantidad] of Object.entries(this.historialCompras)) {
            const producto = catalogo[productoId];
            if (producto && producto.precio) {
                total += producto.precio * cantidad;
            }
        }
        return total;
    }

    // Getters
    getId() {
        return this.id;
    }
    
    getNombre() {
        return this.nombre;
    }
    
    getHistorialCompras() {
        return {...this.historialCompras}; // Devuelve una copia del objeto
    }
}

File: data/TiendaMusical/Cliente.java
import java.util.HashMap;

/**
 * Clase que representa un cliente de la tienda.
 */
public class Cliente {
    private String id;
    private String nombre;
    /**
     * Historial de compras del cliente.
     * Utiliza un HashMap donde la clave es el ID del producto y el valor es la cantidad comprada.
     * Esto permite llevar un registro de los productos comprados y sus cantidades
     */
    private HashMap<String, Integer> historialCompras; // ID Producto -> Cantidad

    public Cliente(String id, String nombre) {
        // TO DO: Implementar el constructor
    }

    /**
     * Añade una compra al historial del cliente.
     * Debe validar que la cantidad sea positiva antes de añadirla.
     * @param productoId ID del producto comprado
     * @param cantidad Cantidad comprada
     */
    public void añadirCompra(String productoId, int cantidad) {
        //TO DO: Implementar el metodo
    }

    /**
     * Calcula el total gastado por el cliente.
     * @param catalogo Catalogo de productos. String es el ID del producto y Producto es la clase que representa al producto.
     *                 Se asume que el catalogo contiene los precios de los productos.
     * @return Total gastado
     */
    public double calcularTotalGastado(HashMap<String, Producto> catalogo) {
        // TODO: Implementar cálculo del total
        return 0.0;
    }

    // Setters y Getters
    public String getId() {
        return id;
    }
    public String getNombre() {
        return nombre;
    }
    public HashMap<String, Integer> getHistorialCompras() {
        return new HashMap<>(historialCompras);
    }
}